### DISCUSSION TITLE
Google authentication with supabase  OAuth

### DISCUSSION THREAD

**Discussion by BryanGabriel-Atangana** [community/community#111401] [Programming Help]
### Select Topic Area

Question

### Body

I'm trying to implement google authentication in my next js app using supabase Auth.
----Error : Uncaught (in promise) undefined
---- Oauth Function ----
 const signInWithGoogle = async () => {
    "use server";
    const supabase = createClient();

    try {
      const { data, error } = await supabase.auth.signInWithOAuth({
        provider: "google",
      });
    } catch (e) {
      console.error("Error signing in with Google:", e);
      return redirect("/login?message=Could not authenticate user");
    }
    return redirect("/protected");
  };
  
  ----- Trigger -----
  <div
          className="border border-foreground/20 rounded-md px-4 py-2 text-foreground mb-2 w-full cursor-pointer"
          onClick={signInWithGoogle}
        >
          <div className="flex flex-row justify-center items-center gap-4 ">
            <div className=" text-[#181818] text-sm">se connecter avec</div>
            <LogoGoogle />
          </div>
        </div>

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @BryanGabriel-Atangana ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
